shared/gateway/gateway_test.go

```diff
@@ -0,0 +1,86 @@
+package gateway
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/prysmaticlabs/prysm/cmd/beacon-chain/flags"
+	"github.com/prysmaticlabs/prysm/shared/testutil/require"
+	logTest "github.com/sirupsen/logrus/hooks/test"
+	"github.com/urfave/cli/v2"
+)
+
+// Test that beacon gateway Start, Stop.
+func TestBeaconGateway_StartStop(t *testing.T) {
+	hook := logTest.NewGlobal()
+
+	app := cli.App{}
+	set := flag.NewFlagSet("test", 0)
+	ctx := cli.NewContext(&app, set, nil)
+
+	gatewayPort := ctx.Int(flags.GRPCGatewayPort.Name)
+	gatewayHost := ctx.String(flags.GRPCGatewayHost.Name)
+	rpcHost := ctx.String(flags.RPCHost.Name)
+	selfAddress := fmt.Sprintf("%s:%d", rpcHost, ctx.Int(flags.RPCPort.Name))
+	gatewayAddress := fmt.Sprintf("%s:%d", gatewayHost, gatewayPort)
+	allowedOrigins := strings.Split(ctx.String(flags.GPRCGatewayCorsDomain.Name), ",")
+	enableDebugRPCEndpoints := ctx.Bool(flags.EnableDebugRPCEndpoints.Name)
+	selfCert := ctx.String(flags.CertFlag.Name)
+
+	beaconGateway := NewBeacon(
+		ctx.Context,
+		BeaconCaller,
+		selfAddress,
+		selfCert,
+		gatewayAddress,
+		nil, /*optional mux*/
+		allowedOrigins,
+		enableDebugRPCEndpoints,
+		ctx.Uint64("grpc-max-msg-size"),
+	)
+
+	beaconGateway.Start()
+	go func() {
+		require.LogsContain(t, hook, "Starting gRPC gateway")
+	}()
+
+	err := beaconGateway.Stop()
```

Comment:
Is there a way to verify that it actually stopped?

Reply:
Unless we add a log like the Start does?

Reply:
Let's keep it the way it is then. It will impact other services, so if we want to add logging it should be in another PR.